From 2d8d7500c9c39719565c815e28d2516e4cbea4dc Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Thu, 26 Mar 2020 00:39:09 +0100
Subject: [PATCH 11/11] ARM: dts: stm32: Add SRAM on FMC2 to STM32MP1 DHCOM

Add bindings for the AS7C34098 SRAM present on the
STM32MP1 DHCOM PDK2.

Signed-off-by: Marek Vasut <marex@denx.de>
---
 arch/arm/boot/dts/stm32mp157c-dhcom-som.dtsi | 23 +++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/stm32mp157c-dhcom-som.dtsi b/arch/arm/boot/dts/stm32mp157c-dhcom-som.dtsi
index 9bd6a300fae6..684c0040392c 100644
--- a/arch/arm/boot/dts/stm32mp157c-dhcom-som.dtsi
+++ b/arch/arm/boot/dts/stm32mp157c-dhcom-som.dtsi
@@ -134,6 +134,22 @@
 			st,fmc2_ebi_cs_bus_turnaround = <9>;
 			st,fmc2_ebi_cs_data_hold = <9>;
 		};
+
+		sram@3,0 {
+			compatible = "mtd-ram";
+			reg = <3 0x0 0x80000>;
+			bank-width = <2>;
+
+			/* Timing values are in nS */
+			st,fmc2_ebi_cs_mux_enable;
+			st,fmc2_ebi_cs_transaction_type = <4>;
+			st,fmc2_ebi_cs_buswidth = <16>;
+			st,fmc2_ebi_cs_address_setup = <6>;
+			st,fmc2_ebi_cs_address_hold = <6>;
+			st,fmc2_ebi_cs_data_setup = <127>;
+			st,fmc2_ebi_cs_bus_turnaround = <9>;
+			st,fmc2_ebi_cs_data_hold = <9>;
+		};
 	};
 };
 
@@ -351,7 +367,8 @@
 				 <STM32_PINMUX('D', 8, AF12)>, /* FMC_D13 */
 				 <STM32_PINMUX('D', 9, AF12)>, /* FMC_D14 */
 				 <STM32_PINMUX('D', 10, AF12)>, /* FMC_D15 */
-				 <STM32_PINMUX('G', 9, AF12)>; /* FMC_NE2_FMC_NCE */
+				 <STM32_PINMUX('G', 9, AF12)>, /* FMC_NE2_FMC_NCE */
+				 <STM32_PINMUX('G', 12, AF12)>; /* FMC_NE4 */
 			bias-disable;
 			drive-push-pull;
 			slew-rate = <3>;
@@ -379,10 +396,10 @@
 				 <STM32_PINMUX('D', 8, ANALOG)>, /* FMC_D13 */
 				 <STM32_PINMUX('D', 9, ANALOG)>, /* FMC_D14 */
 				 <STM32_PINMUX('D', 10, ANALOG)>, /* FMC_D15 */
-				 <STM32_PINMUX('G', 9, ANALOG)>; /* FMC_NE2_FMC_NCE */
+				 <STM32_PINMUX('G', 9, ANALOG)>, /* FMC_NE2_FMC_NCE */
+				 <STM32_PINMUX('G', 12, ANALOG)>; /* FMC_NE4 */
 		};
 	};
-
 };
 
 &pwr_regulators {
-- 
2.25.1

